x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -158,6 +172,58 @@ export const postActionsConnectorExecuteRoute = (
               });
             }
 
+            if (conversation?.title === 'New chat' && prevMessages) {
+              try {
+                const autoTitle = await executeAction({
+                  actions,
+                  request,
+                  connectorId,
+                  actionTypeId: connectors[0]?.actionTypeId,
+                  params: {
+                    subAction: request.body.subAction,
+                    subActionParams: {
+                      model: request.body.model,
+                      messages: [
+                        {
+                          role: 'assistant',
+                          content:
+                            'You are a helpful assistant for Elastic Security. Assume the following message is the start of a conversation between you and a user; give this conversation a title based on the content below. DO NOT UNDER ANY CIRCUMSTANCES wrap this title in single or double quotes. This title is shown in a list of conversations to the user, so title it for the user, not for you.',
```

Comment:
Hmm.. should this be in `i18n`? I think when we use LangChain, we can use English, but since this is creating the title, we should match the user's language

Reply:
Good point! Will internationalize this. 👍 

Tangentially, we should probably go through a multi-lingual test cycle to get a baseline for the non-english UX. Since we only store the ES|QL docs in english, I'm not sure how ELSER will perform on retrieval using query embeddings in a different language.

Reply:
Agreed. Maybe we should bring in @jamesspi, could you please help to facilitate this testing?

Reply:
Also adding `i18n` for `if (conversation?.title === 'New chat' && prevMessages) {` as `New chat` is internationalized client side, so autoTitle functionality won't work in different languages.